dataactcore/utils/jobQueue.py

```diff
@@ -22,6 +30,32 @@ def enqueue(jobID):
     return response.json()
 
 
+def generate_f_file(submission_id, job_id, interface_holder_class,
+                    timestamped_name, is_local):
+    """Write rows from fileF.generateFRows to an appropriate CSV. Here the
+    third parameter, interface_holder_class, is a bit of a hack. Importing
+    InterfaceHolder directly causes cyclic dependency woes, so we're passing
+    in a class"""
+    with job_app.app_context():
+        job_manager = interface_holder_class().jobDb
+
+        try:
+            rows_of_dicts = fileF.generateFRows(job_manager.session,
+                                                submission_id)
+            header = [key for key in fileF.mappings]    # keep order
+            body = []
+            for row in rows_of_dicts:
+                body.append([row[key] for key in header])
+
+            write_csv(timestamped_name, is_local, header, body)
+            job_manager.markJobStatus(job_id, "finished")
```

Comment:
@jworcestBAH Thanks for weighing in here... all good ideas and concerns.

Re: changing everything to static functions in their present locations...my .02 is that the current organization makes the functions hard to find because they are scattered between the broker, core, and validator. I did toss around the idea of creating a single file of static functions for each of the old db names and putting those in `core/interfaces`. 

When I started playing with that in practice, I realized that it doesn't solve the circular reference issue, because the separate files would all still have to import each other. I also come back to that fact that these names (_e.g._, error, job, etc.) aren't meaningful now that we've consolidated the databases, so I don't see an advantage to keeping that distinction in the code base.

I *really* agree with you that a big old file with a bunch of database functions is not a good final goal. I see it as a starting point--a common place to hold everything as we get it the individual pieces moved out to the models, replaced with direct queries, celery dependencies, etc.

Thanks for pointing out the additional logic embedded in `markJobStatus`. That's something I missed when scanning the code.